src/main/java/RecommendationsServlet.java

```diff
@@ -0,0 +1,184 @@
+import com.google.gson.JsonObject;
+import com.google.gson.Gson;
+import com.google.sps.data.VisualizationData;
+import com.google.sps.data.History;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.lang.String;
+import java.util.Random;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+@WebServlet("/recommendations")
+public class RecommendationsServlet extends HttpServlet {
+  private ArrayList<String> personType = new ArrayList<String>(
+    Arrays.asList("under-18", "over-18", "all-ages", "male", "female"));
+  private ArrayList<String> action = new ArrayList<String>(
+    Arrays.asList("live", "work", "moved"));
+  private ArrayList<String> location = new ArrayList<String>(
+    Arrays.asList("state", "01", "02", "04", "05", "06", "08",
+    "09", "10", "11", "12", "13", "15", "16", "17", "18", "19",
+    "20", "21", "22", "23", "24", "25", "26", "27", "28", "29",
+    "30", "31", "32", "33", "34", "35", "36", "37", "38", "39",
+    "40", "41", "42", "44", "45", "46", "47", "48", "49", "50",
+    "51", "53", "54", "55", "56"));
+  private ArrayList<String> years = new ArrayList<String> (
+    Arrays.asList("2010", "2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018")); 
+  private HashMap<String, Integer> personTypeMap = new HashMap<String, Integer>();
+  private HashMap<String, Integer> actionMap = new HashMap<String, Integer>();
+  private HashMap<String, Integer> locationMap = new HashMap<String, Integer>(); 
+  private HashMap<String, Integer> yearsMap = new HashMap<String, Integer>(); 
+
+  // if the user doesn't have any history to draw from to get recommendations, 
+  // return default recommendations that would be applicable to most users which is
+  // the number of children, adults, people, men, and women who lived in each 
+  // US state in 2018. 
+  private ArrayList<VisualizationData> getDefaultRecommendations(String userId) {
+    ArrayList<VisualizationData> defaultRecommendations = new ArrayList<VisualizationData>();
+    for (String person: personType) {
+      defaultRecommendations.add(new VisualizationData(userId, person, "live", "state", "2018"));
+    }
+    return defaultRecommendations;
+  }
+
+  // uses java.util.random to get a random integer between 
+  // 0 and upperBound-1 in order to access a random element 
+  // for each of the options
+  private int getRandomInt(int upperBound) {
+    Random randomObject = new Random();
+    int randomInt = randomObject.nextInt(upperBound);
+    return randomInt;
```

Comment:
Can be one line - return new Random().nextInt(upperBound);

Reply:
fixed.